tangelo/toolboxes/post_processing/histogram.py

```diff
@@ -0,0 +1,225 @@
+
+        Args:
+            expected_outcomes (dict): Wanted outcomes on certain qubit indices and
+                their expected state. For example, {0: "1"} would filter results
+                based on the first qubit with the 1 state measured. This argument
+                can also filter many qubits.
+        """
+        def f_post_select(bitstring):
+            for qubit_i, expected_bit in expected_outcomes.items():
+                if bitstring[qubit_i] != expected_bit:
+                    return False
+            return True
+
+        new_hist = filter_hist(self, f_post_select)
+        self.counts = new_hist.counts
+        self.remove_bits(*list(expected_outcomes.keys()))
+
+    def resample(self, n_shots):
+        """Post selection is done with the tangelo.toolboxes.post_processing.boostrapping.get_resampled_frequencies
+        function (see the relevant documentation for more details).
+
+        Args:
+            n_shots (int): Self-explanatory.
+
+        Returns:
+            Histogram: resampled data with n_shots from the distribution.
+        """
+        new_frequencies = get_resampled_frequencies(self.frequencies, n_shots)
+        return Histogram(new_frequencies, n_shots)
+
+    def get_expectation_value(self, term, coeff=1.):
+        """Output the expectation value for qubit operator term. The histogram
+        data is expected to have been acquired in a compatible measurement
+        basis.
+
+        Args:
+            term(openfermion-style QubitOperator object): a qubit operator, with
+                only a single term.
+            coeff (imaginary): Coefficient to multiply the eigenvalue by.
+
+        Returns:
+            imaginary: Expectation value for this operator.
+        """
+        return coeff*Simulator.get_expectation_value_from_frequencies_oneterm(term, self.frequencies)
+
+
+def aggregate_histograms(*hists):
+    """Return a Histogram object formed from data aggregated from all input
+    Histogram objects.
+
+    Args:
+        hists (variable number of Histogram objects): the input Histogram objects.
+
+    Returns:
+        Histogram: The aggregated histogram.
+    """
+
+    if not hists:
+        raise ValueError(f"{aggregate_histograms.__name__} takes at least one Histogram object as an argument")
+    elif len(hists) == 1:
+        return hists[0]
+    else:
+        # Check that data is on the same number of qubits (same bitstring lengths).
+        lengths_bitstrings = set([h.n_qubits for h in hists])
+        if len(lengths_bitstrings) > 1:
+            raise ValueError(f"Input histograms have different bitstring lengths ({lengths_bitstrings})")
+
+        # Aggregate histograms.
+        total_counter = sum([Counter({k: v for k, v in h.counts.items()}) for h in hists], Counter())
+
+        return Histogram(dict(total_counter))
+
+
+def filter_hist(hist, function, *args, **kwargs):
+    """Filter selection function to consider bistrings in respect to a boolean
```

Comment:
Btw such function is called a predicate.

Here's how in practice people have been doing it with list comprehension, and in older python with `filter`: https://www.geeksforgeeks.org/filter-python-list-by-predicate-in-python/